OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/ResetPasswordDialogFragment.java

```diff
@@ -116,75 +121,60 @@ private void submit(@Nullable String email) {
         } else if (!InputValidationUtil.isValidEmail(email)) {
             showError(getString(R.string.error_invalid_email));
         } else {
-            loginService.resetPassword(email).enqueue(new ErrorHandlingCallback<ResetPasswordResponse>(
+            setUiForInteraction(false);
+            loadingIndicator.setVisibility(View.VISIBLE);
+            resetCall = loginService.resetPassword(email);
+            resetCall.enqueue(new ErrorHandlingCallback<ResetPasswordResponse>(
                     getContext(), CallTrigger.USER_ACTION) {
                 @Override
                 protected void onResponse(@NonNull final ResetPasswordResponse result) {
+                    setUiForInteraction(true);
+                    final OnActivityResultListener listener =
+                            OnActivityResultListener.Util.getListener(getContext());
                     if (listener != null) {
                         if (result.isSuccess()) {
-                            listener.onActivityResult(0, Activity.RESULT_OK, null);
+                            listener.onActivityResult(0, RESULT_OK, null);
                             dismiss();
                         } else {
                             final String errorMsg = result.getPrimaryReason();
                             final Intent data = new Intent();
                             data.putExtra("error", errorMsg);
-                            listener.onActivityResult(0, Activity.RESULT_CANCELED, data);
+                            listener.onActivityResult(0, RESULT_FAILURE, data);
                             showError(errorMsg);
                         }
                     }
                 }
 
                 @Override
                 protected void onFailure(@NonNull Throwable error) {
+                    setUiForInteraction(true);
+                    final OnActivityResultListener listener =
+                            OnActivityResultListener.Util.getListener(getContext());
                     if (listener != null) {
                         final String errorMsg = ErrorUtils.getErrorMessage(error, getContext());
                         final Intent data = new Intent();
                         data.putExtra("error", errorMsg);
-                        listener.onActivityResult(0, Activity.RESULT_CANCELED, data);
+                        listener.onActivityResult(0, RESULT_FAILURE, data);
```

Comment:
As I [mentioned][1] previously, you shouldn't return a result here since the dialog is still active and not dismissed. Nor do you need to define your own result constants; the ones defined in `Activity` are sufficient. You should just override `onCancel()` instead, and return `RESULT_CANCELED` from there.

[1]: https://github.com/edx/edx-app-android/pull/877#discussion_r88229891